http-service/src/main/java/net/runelite/http/service/hiscore/HiscoreService.java

```diff
@@ -67,53 +102,74 @@ private HiscoreResultBuilder lookupUsername(String username) throws IOException
 			.build();
 
 		Response okresponse = RuneliteAPI.CLIENT.newCall(okrequest).execute();
-		String responseStr;
 
-		try (ResponseBody body = okresponse.body())
+		if (!okresponse.isSuccessful())
 		{
-			responseStr = body.string();
+			switch (HttpStatus.valueOf(okresponse.code()))
+			{
+				case NOT_FOUND:
+					throw new NotFoundException();
+				default:
+					throw new RuntimeException("Error retrieving data from Jagex Hiscores: " + okresponse.message());
+			}
 		}
+		else
```

Comment:
You don't need this else or have to indent everything below it because control cannot flow out of the if